File: FrontEnd/my-react-app/src/services/userService.js
import axiosClient from './axiosClient';

export const userService = {
  getAllUsers: (params) => {
    return axiosClient.get('/users', { params });
  },
  getUserById: (id) => {
    return axiosClient.get(`/users/${id}`);
  },
  createUser: (data) => {
    return axiosClient.post('/users', data);
  },
  updateUser: (id, data) => {
    return axiosClient.put(`/users/${id}`, data);
  },
  deleteUser: (id) => {
    return axiosClient.delete(`/users/${id}`);
  }
};

File: Backend/src/main/java/com/InterviewHub/feature/user/UserService.java
package com.InterviewHub.feature.user;

import com.InterviewHub.dto.PaginationDTO;
import com.InterviewHub.feature.user.dto.CreateUserRequest;
import com.InterviewHub.feature.user.dto.UpdateUserRequest;
import com.InterviewHub.feature.user.dto.UserResponse;

public interface UserService {
    UserResponse createUser(CreateUserRequest request);
    UserResponse getUserById(Long id);
    PaginationDTO<UserResponse> getAllUsers(int page, int size);
    UserResponse updateUser(Long id, UpdateUserRequest request);
    void deleteUser(Long id);
}
